lib/spack/spack/build_systems/sourceforge.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import spack.util.url
+import spack.package
+
+
+class SourceforgePackage(spack.package.PackageBase):
+    """Mixin that takes care of setting url and mirrors for Sourceforge
+       packages."""
+    #: Path of the package in a Sourceforge mirror
+    sourceforge_mirror_path = None
+
```

Comment:
@alalazo we could potentially try to sneak in a `list_url` in these mixin classes, I guess.